### ISSUE TITLE
Param to name route component during `ng g m`

### ISSUE THREAD

**Issue by moniuch** [angular/angular-cli#21822]
Every time I use this sort of syntax:
`ng g m board --routing true --module app.module --route board`
I find myself renaming the route component I got from schematics.

The part `--route board` automatically creates a route component that is named exactly as the route, in this case `board.component.*`. Meanwhile, my practice is to have route components named with `page` in name, so in this case it would be `board-page.component.*` It would have its own router outlet where I would display containers (smart components) - one of them would probably be `board.component.*`

Could be I am not the only one?

Couldn't cli allow a param to name the component as well, say `--component pages/board-page`
If not supplied, `ng g m` would behave as it does currently.

**Comment by alan-agius4** [comment]
This is quite a custom use-case. 

I did see some "page" patterns in the past, Typically this however resulted in changing the component suffix type and not actually changing the feature name. Ex: `profile-desktop.page.ts`.  This is also similar to containers, where they are typically suffixed with `.container` instead of `.component`. example, `profile-desktop.container.ts`.

I don't think that adding another option in the module to pass it down to the component schematic isn't something that we'd like to do, especially considering there isn't really much interest in this from the community since this feature was implemented. This also can be archived by extending the default schematic, which is really the way to go for these sort of custom style-guides.
